ts/store/reducers/entities/messages/thirdPartyById.ts

```diff
@@ -39,14 +46,49 @@ export const thirdPartyByIdReducer = (
   return state;
 };
 
+const thirdPartyKeyValueContainer = (state: GlobalState) =>
+  state.entities.messages.thirdPartyById;
+
 /**
  * From UIMessageId to the third party content pot
  */
 export const thirdPartyFromIdSelector = createSelector(
   [
-    (state: GlobalState) => state.entities.messages.thirdPartyById,
-    (_: GlobalState, id: UIMessageId) => id
+    thirdPartyKeyValueContainer,
+    (_: GlobalState, ioMessageId: UIMessageId) => ioMessageId
   ],
-  (byId, id): pot.Pot<ThirdPartyMessageWithContent, Error> =>
-    byId[id] ?? pot.none
+  (
+    thirdPartyKeyValueContainer,
+    ioMessageId
+  ): pot.Pot<ThirdPartyMessageWithContent, Error> =>
+    thirdPartyKeyValueContainer[ioMessageId] ?? pot.none
 );
+
+export const thirdPartyMessageUIAttachment =
+  (state: GlobalState) =>
+  (ioMessageId: UIMessageId) =>
+  (thirdPartyMessageAttachmentId: UIAttachmentId) =>
+    pipe(
+      thirdPartyFromIdSelector(state, ioMessageId),
+      pot.toOption,
+      O.map(
+        thirdPartyMessage => thirdPartyMessage.third_party_message.attachments
+      ),
+      O.map(thirdPartyMessageAttachments =>
+        thirdPartyMessageAttachments?.find(
+          thirdPartyMessageAttachment =>
+            thirdPartyMessageAttachment.id ===
+            (thirdPartyMessageAttachmentId as string as NonEmptyString)
+        )
+      ),
+      O.map(thirdPartyMessageAttachment =>
+        thirdPartyMessageAttachment
+          ? attachmentFromThirdPartyMessage(
+              ioMessageId,
+              thirdPartyMessageAttachment,
+              "GENERIC"
+            )
+          : null
+      ),
+      O.toUndefined
+    );
```

Comment:
when there are optional values we can use `chainNullableK`.
what do you think?

```suggestion
    pipe(
      thirdPartyFromIdSelector(state, ioMessageId),
      pot.toOption,
      O.chainNullableK(
        thirdPartyMessage => thirdPartyMessage.third_party_message.attachments
      ),
      O.chainNullableK(attachments =>
        attachments.find(
          a =>
            a.id === (thirdPartyMessageAttachmentId as string as NonEmptyString)
        )
      ),
      O.fold(
        () => undefined,
        attachment =>
          attachmentFromThirdPartyMessage(ioMessageId, attachment, "GENERIC")
      )
    );
```

Reply:
Refactored 🥳